flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/ChangelogStorageMetricGroup.java

```diff
@@ -0,0 +1,134 @@
+        this.uploadsCounter = counter(CHANGELOG_STORAGE_UPLOAD_REQUESTS, new ThreadSafeCounter());
+        this.logsPerUpload =
+                histogram(
+                        CHANGELOG_STORAGE_LOGS_PER_UPLOAD,
+                        new DescriptiveStatisticsHistogram(WINDOW_SIZE));
+        this.attemptsPerUpload =
+                histogram(
+                        CHANGELOG_STORAGE_ATTEMPTS_PER_UPLOAD,
+                        new DescriptiveStatisticsHistogram(WINDOW_SIZE));
+        this.uploadSizes =
+                histogram(
+                        CHANGELOG_STORAGE_UPLOAD_SIZES,
+                        new DescriptiveStatisticsHistogram(WINDOW_SIZE));
+        this.uploadLatencies =
+                histogram(
+                        CHANGELOG_STORAGE_UPLOAD_LATENCIES,
+                        new DescriptiveStatisticsHistogram(WINDOW_SIZE));
+        this.uploadFailures = counter(CHANGELOG_STORAGE_UPLOAD_FAILURES, new ThreadSafeCounter());
+    }
+
+    public Counter getUploadsCounter() {
+        return uploadsCounter;
+    }
+
+    public Counter getFailuresCounter() {
+        return uploadFailures;
+    }
+
+    public Histogram getAttemptsPerUpload() {
+        return attemptsPerUpload;
+    }
+
+    public Histogram getLogsPerUpload() {
+        return logsPerUpload;
+    }
+
+    public Histogram getUploadSizes() {
+        return uploadSizes;
+    }
+
+    public Histogram getUploadLatencies() {
+        return uploadLatencies;
+    }
+
+    public void registerQueueSizeGauge(Gauge<Integer> gauge) {
+        gauge(CHANGELOG_STORAGE_QUEUE_SIZE, gauge);
+    }
+
+    private static class ThreadSafeCounter implements Counter {
+        private final LongAdder longAdder = new LongAdder();
+
+        @Override
+        public void inc() {
+            longAdder.increment();
+        }
+
+        @Override
+        public void inc(long n) {
+            longAdder.add(n);
+        }
+
+        @Override
+        public void dec() {
+            longAdder.decrement();
+        }
+
+        @Override
+        public void dec(long n) {
+            longAdder.add(-n);
+        }
+
+        @Override
+        public long getCount() {
+            return longAdder.longValue();
+        }
+    }
+
+    private static final String PREFIX = "changelogStorage";
+    public static final String CHANGELOG_STORAGE_UPLOAD_REQUESTS = PREFIX + ".uploadRequests";
+    public static final String CHANGELOG_STORAGE_UPLOAD_FAILURES = PREFIX + ".uploadFailures";
+    public static final String CHANGELOG_STORAGE_UPLOAD_SIZES = PREFIX + ".uploadSizes";
+    public static final String CHANGELOG_STORAGE_UPLOAD_LATENCIES = PREFIX + ".uploadLatencies";
+    public static final String CHANGELOG_STORAGE_ATTEMPTS_PER_UPLOAD =
+            PREFIX + ".attemptsPerUpload";
+    public static final String CHANGELOG_STORAGE_LOGS_PER_UPLOAD = PREFIX + ".logsPerUpload";
+    public static final String CHANGELOG_STORAGE_QUEUE_SIZE = PREFIX + ".queueSize";
```

Comment:
"Pending" seems ambiguous to me because uploads to retry and pre-emptive data to upload can also be called as such.

And "current" seems redundant, how about `uploadQueueSize`?